src/file/properties.rs

```diff
@@ -0,0 +1,550 @@
+      .or_else(|| self.default_column_properties.max_statistics_size())
+      .unwrap_or(DEFAULT_MAX_STATISTICS_SIZE)
+  }
+}
+
+/// Writer properties builder.
+pub struct WriterPropertiesBuilder {
+  data_pagesize_limit: usize,
+  dictionary_pagesize_limit: usize,
+  write_batch_size: usize,
+  max_row_group_size: usize,
+  writer_version: WriterVersion,
+  created_by: String,
+  default_column_properties: ColumnProperties,
+  column_properties: HashMap<ColumnPath, ColumnProperties>
+}
+
+impl WriterPropertiesBuilder {
+  /// Returns default state of the builder.
+  fn with_defaults() -> Self {
+    Self {
+      data_pagesize_limit: DEFAULT_PAGE_SIZE,
+      dictionary_pagesize_limit: DEFAULT_DICTIONARY_PAGE_SIZE_LIMIT,
+      write_batch_size: DEFAULT_WRITE_BATCH_SIZE,
+      max_row_group_size: DEFAULT_MAX_ROW_GROUP_SIZE,
+      writer_version: DEFAULT_WRITER_VERSION,
+      created_by: DEFAULT_CREATED_BY.to_string(),
+      default_column_properties: ColumnProperties::new(),
+      column_properties: HashMap::new()
+    }
+  }
+
+  /// Finalizes the configuration and returns immutable writer properties struct.
+  pub fn build(self) -> WriterProperties {
+    WriterProperties {
+      data_pagesize_limit: self.data_pagesize_limit,
+      dictionary_pagesize_limit: self.dictionary_pagesize_limit,
+      write_batch_size: self.write_batch_size,
+      max_row_group_size: self.max_row_group_size,
+      writer_version: self.writer_version,
+      created_by: self.created_by,
+      default_column_properties: self.default_column_properties,
+      column_properties: self.column_properties
+    }
+  }
+
+  // ----------------------------------------------------------------------
+  // Writer properies related to a file
+
+  /// Sets writer version.
+  pub fn set_writer_version(mut self, value: WriterVersion) -> Self {
+    self.writer_version = value;
+    self
+  }
+
+  /// Sets data page size limit.
+  pub fn set_data_pagesize_limit(mut self, value: usize) -> Self {
+    self.data_pagesize_limit = value;
+    self
+  }
+
+  /// Sets dictionary page size limit.
+  pub fn set_dictionary_pagesize_limit(mut self, value: usize) -> Self {
+    self.dictionary_pagesize_limit = value;
+    self
+  }
+
+  /// Sets write batch size.
+  pub fn set_write_batch_size(mut self, value: usize) -> Self {
+    self.write_batch_size = value;
+    self
+  }
+
+  /// Sets max size for a row group.
+  pub fn set_max_row_group_size(mut self, value: usize) -> Self {
+    self.max_row_group_size = value;
+    self
+  }
+
+  /// Sets "created by" property.
+  pub fn set_created_by(mut self, value: String) -> Self {
+    self.created_by = value;
+    self
+  }
+
+  // ----------------------------------------------------------------------
+  // Setters for any column (global)
+
+  /// Sets (fallback) encoding for any column.
+  pub fn set_encoding(mut self, value: Encoding) -> Self {
+    self.default_column_properties.set_encoding(value);
```

Comment:
Should we check if `value` is dictionary encoding here? what if people pass RLE_DICTIONARY as `value` and then enable dictionary encoding for some column?

Reply:
Good point! This is a bug, I will test for it. Yes, we should have some place where we check all of the configurations. Will update.

Reply:
I will add all checks when I build `WriterProperties` - that should solve the problem.

Reply:
I looked at the code again. It seems that we do not allow setting RLE_DICTIONARY or PLAIN_DICTIONARY, regardless of whether a dictionary is enabled or not, so someone wants to enable dictionary encoding they should use another method. I will add a test for this situation though.

Reply:
Sounds good.